api/src/wfl/api/handlers.clj

```diff
@@ -102,9 +103,17 @@
   (log/info (select-keys request [:request-method :uri :parameters]))
   (let [uuid   (get-in request [:path-params :uuid])
         status (get-in request [:body-params :status])]
+    ;; TODO: Refactor to workflow-status whitelist
+    ;; https://broadinstitute.atlassian.net/browse/GH-1424
+    (when (= "Succeeded" status)
+      (throw (UserException. "Retry unsupported for workflow status."
+                             {:uuid            uuid
+                              :workflow-status status
+                              :status          400})))
     (->> (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-           (let [w (workloads/load-workload-for-uuid tx uuid)]
-             [w (workloads/workflows-by-status tx w status)]))
+           (let [workload  (workloads/load-workload-for-uuid tx uuid)
+                 workflows (workloads/workflows-by-status tx workload status)]
+             [workload workflows]))
```

Comment:
I wanted to throw here if we couldn't find any workflows for the status, but I had some trouble adjusting the system tests accordingly for pipelines without retry implemented.  So I settled for a check on this in the COVID implementation.